Sources/TuistSupport/MachineEnvironment.swift

```diff
@@ -15,15 +15,29 @@ public class MachineEnvironment: MachineEnvironmentRetrieving {
     private init() {}
 
     /// `clientId` is a unique anonymous hash that identifies the machine running Tuist
-    public let clientId = (Host.current().name?.checksum(algorithm: .md5)) ?? "unknown"
+    public let clientId = (
+        try? System.shared
+            .capture("ioreg", "-d2", "-c", "IOPlatformExpertDevice")
+            .components(separatedBy: "\"IOPlatformUUID\" = \"")
+            .last?.components(separatedBy: "\"")
+            .first?.checksum(algorithm: .md5)
+    ) ?? "unknown"
 
     /// The `macOSVersion` of the machine running Tuist, in the format major.minor.path, e.g: "10.15.7"
-
-    // swiftlint:disable line_length
-    public let macOSVersion = "\(ProcessInfo.processInfo.operatingSystemVersion.majorVersion).\(ProcessInfo.processInfo.operatingSystemVersion.minorVersion).\(ProcessInfo.processInfo.operatingSystemVersion.patchVersion)"
+    public let macOSVersion = """
```

Comment:
I recommend lazily caching the properties in this class because they don't change while Tuist is being executed. I think we can place the caching logic in a property wrapper that we can bind to the property:

```swift
@threadSafeCached
public let macOSVersion: String

Reply:
Any reason why we can't just use `lazy var`?

Reply:
Lazy initialization and thread safety are two different things. We want lazy initialization, but also that multiple threads can access the value without leading to race conditions.